src/core/name_number_tree.js

```diff
@@ -135,12 +135,17 @@ class NameOrNumberTree {
         } else if (key > currentKey) {
           l = m + 2;
         } else {
-          return xref.fetchIfRef(entries[m + 1]);
+          return entries[m + 1];
         }
       }
     }
     return null;
   }
+
+  get(key) {
+    const ref = this.getRaw(key);
+    return (ref && this.xref.fetchIfRef(ref)) || null;
```

Comment:
It's not clear to me that this is generally safe/correct, since it'd not work as intended if the value would ever be e.g. an empty string or zero.
```suggestion
    const raw = this.getRaw(key);
    return raw instanceof Ref ? this.xref.fetch(raw) : raw;
```